modules/seqimporter/src/main/scala/gem/seqimporter/SequenceImporter.scala

```diff
@@ -0,0 +1,152 @@
+
+      // Drop the preliminary cruft and then parse the sequence XML.  We're
+      // assuming that the observation was found in the ODB and all went well
+      // or else this will fail with an exception.  We could fish out XML-RPC
+      // failure results and return more accurate failure information if needed.
+      XML.loadString(xmlTxt.lines.drop(3).mkString("\n"))
+    }
+  }
+
+  /** Converts the OCS2 sequence XML into the new sequence model.  Each step
+    * includes the instrument configuration and its completion state.
+    */
+  private def parseSequenceXml(oid: Observation.Id, xml: Elem): List[Step[(InstrumentConfig, Boolean)]] = {
+    // Extracts the steps in the XML sequence to a simple list of Maps where
+    // each Map is all the keys and values that apply to the step.
+    def configMaps(xml: Elem): List[ConfigMap] = {
+      def extractSystem(configMap: ConfigMap, sys: Node): ConfigMap = {
+        val sysName = (sys \ "@name").text
+
+        (configMap /: (sys \ "param")) { (map, param) =>
+          val paramName  = (param \ "@name").text
+          val paramValue = (param \ "@value").text
+          map + (s"$sysName:$paramName" -> paramValue)
+        }
+      }
+
+      def extractStep(configMap: ConfigMap, step: Node): ConfigMap =
+        (configMap /: (step \ "system")) { extractSystem }
+
+      (xml \ "step").toList.scanLeft(EmptyConfigMap) { extractStep }.tail
+    }
+
+    configMaps(xml).map(ConfigReader.unsafeStep)
+  }
+
+  /** Fetches a Sequence from the ODB using its WDBA service, translating it
+    * into the new sequence model in memory.
+    */
+  def fetchSequence(oid: Observation.Id): Task[List[Step[(InstrumentConfig, Boolean)]]] =
+    fetchSequenceXml(oid).map { parseSequenceXml(oid, _) }
+
+  private val configureLogging: Task[Unit] =
+    Task.delay(List(
+      "edu.gemini.spModel.type.SpTypeUtil"
+    ).map(Logger.getLogger).foreach(_.setLevel(Level.OFF)))
+
+  // We need a minimal program/observation in order to insert steps.  The
+  // seqexec doesn't actually use these but they have to be there to satisfy
+  // database constraints.
+  private def createMinimalContext(oid: Observation.Id, s: List[Step[(InstrumentConfig, Boolean)]]): Task[Unit] = {
+    val i = s.headOption.map(_.instrument._1.instrument)
+    val p = Program[Nothing](oid.pid, "", Nil)
+    val o = Observation[Nothing](oid, "", i, Nil)
+
+    (for {
+      _ <- ProgramDao.insert(p)
+      _ <- ObservationDao.insert(o)
+    } yield ()).transact(lxa)
+  }
+
+  // Writes the sequence to the database.
+  private def writeSequence(oid: Observation.Id, s: List[Step[(InstrumentConfig, Boolean)]]): Task[Unit] = {
+    // Create datasets for the complete steps. Fake filename and timestamp.
+    // Seqexec doesn't need to read these values anyway.
+    val datasets = s.map(_.instrument._2).zipWithIndex.map { case (complete, i) =>
+      complete option Dataset(Dataset.Label(oid, i+1), s"unused", Instant.now)
```

Comment:
The labels are real, just as they are in the ODB.  It's the other information that I don't have access to looking just at the sequence XML.  To get the actual filename and timestamp I'd need to have access to the observation's obslog node.  I think we should create a new OCS2 service that provides the full program/observation configuration along with its unrolled sequence.  At that point, I can populate the `dataset` table with all real information.  Even now though, the seqexec could read this table to determine whether a step is complete.